### PR TITLE
Log the resource type associated with a download job

### PR DESCRIPTION
**Pull request by whytheplatypus**
- starts a pattern for adding fields to the contextual log (the log associated with the request).
- using middleware to decorate functions with additional request level logging.

### Proposed Changes

- [x] Moving to structured logging
- [x] Upping test coverage
- [x] Experimenting with CI pipeline checks

### Change Details

Using the structured contextual logging provided by the chi middleware

### Security Implications
None: adding only one non-pii/phi field to logging.

- [ ] new software dependencies
None

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [ ] no PHI/PII is affected by this change

### CODE REVIEW

**Review by 4ell0 — APPROVED** [review]
Looks good

File: bcda/logging/middleware.go [hunk 1]
```diff
@@ -83,6 +86,48 @@ func (l *StructuredLoggerEntry) Panic(v interface{}, stack []byte) {
 	})
 }
 
+type ResourceTypeLogger struct {
+	Repository models.JobKeyRepository
+}
+
+func (rl *ResourceTypeLogger) LogJobResourceType(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		next.ServeHTTP(w, r)
```
Comment (ericbuckley) [inline bcda/logging/middleware.go:95]: Am I reading this correctly that only a 200 response from `v1.ServeData` will result in the resource type being logged?  Not necessarily a bad thing, just wanted to make sure I was interpreting this middleware function correctly.

**Review by ericbuckley — APPROVED** [review]
looks great, thanks for moving this into middleware

File: bcda/logging/middleware.go (hunk 1 above)
Comment (whytheplatypus) [reply bcda/logging/middleware.go:95]: This should run regardless, although if ServeData isn't returning a 200 it's likely that the same thing that caused that will mean there's no ResourceType to log (e.g. the job id was wrong, or something)

**Review by ian-a-sawyer — APPROVED** [review]
Looks great!

**Review by KyRobbins — APPROVED** [review]

**Pull request merged**